src/App.jsx

```diff
@@ -14,57 +15,69 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const handleAddButtonClick = good => {
+    setSelectedGood(good);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  const handleRemoveButtonClick = () => {
+    setSelectedGood(null);
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  return (
+    <main className="section container">
+      {!selectedGood ? (
+        <h1 className="title is-flex is-align-items-center">
+          No goods selected
+        </h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood} is selected
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={handleRemoveButtonClick}
+          />
+        </h1>
+      )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={good}
+              data-cy="Good"
+              className={
+                selectedGood === good && 'has-background-success-light'
+              }
             >
-              -
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={selectedGood === good ? 'RemoveButton' : 'AddButton'}
+                  type="button"
+                  className={selectedGood === good && 'button is-info'}
+                  onClick={() => {
+                    if (selectedGood === good) {
+                      handleRemoveButtonClick();
+                    } else {
+                      handleAddButtonClick(good);
+                    }
+                  }}
+                >
+                  {selectedGood === good ? '-' : '+'}
+                </button>
```

Comment:
It would be better to conditionally render the entire button, rather than its separate components
{selectedGood === good ? (
                  <button
                    ...
                    onClick={() => {
                      setSelectedGood('');
                    }}
                  >
                    -
                  </button>
                ) : (
                  ...
                  <button
                    onClick={() => {
                      setSelectedGood(good);
                    }}
                  >
                    +
                  </button>
                )}